src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -570,15 +570,17 @@ internal void ReadToken(ReadOnlyMemory<char> encodedTokenMemory)
         internal JsonClaimSet CreateClaimSet(ReadOnlySpan<char> strSpan, int startIndex, int length, bool createHeaderClaimSet)
         {
             int outputSize = Base64UrlEncoding.ValidateAndGetOutputSize(strSpan, startIndex, length);
+
             byte[] output = ArrayPool<byte>.Shared.Rent(outputSize);
             try
             {
-                Base64UrlEncoding.Decode(strSpan, startIndex, length, output);
+                ReadOnlySpan<char> slice = strSpan.Slice(startIndex, length);
+                Base64UrlEncoder.UnsafeDecode(slice, output);
```

Comment:
`Base64UrlEncoding.Decode(..)` was throwing ArgumentOutOfRangeException and ArgumentException exceptions. The new code path will not throw these exceptions. We might want to share this in the release notes.

Reply:
yes, good idea.